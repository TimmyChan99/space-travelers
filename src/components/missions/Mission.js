import React from 'react';
import PropTypes from 'prop-types';

export default function Mission({ mission }) {
  const handleJoinMission = (id) => {
    console.log(id);
  };
  const handleLeaveMission = (id) => {
    console.log(id);
  };

  const status = mission.status ? (
    <div className="active-mb"> Active Member</div>
  ) : (
    <div className="no-active-mb"> NOT A MEMBER</div>
  );

  const missionBt = !mission.status ? (
    <button
      type="button"
      className="join-mission-bt"
      onClick={() => handleJoinMission(mission.missionId)}
    >
      Join Mission
    </button>
  ) : (
    <button
      type="button"
      className="leave-mission-bt"
      onClick={() => handleLeaveMission(mission.missionId)}
    >
      Leave Mission
    </button>
  );

  return (
    <div className="mission-container">
      <div className="mission-name">{mission.missionName}</div>
      <div className="mission-description">{mission.description}</div>
      {status}
      {missionBt}
    </div>
  );
}

Mission.propTypes = {
  mission: PropTypes.oneOfType([PropTypes.object]).isRequired,
};
