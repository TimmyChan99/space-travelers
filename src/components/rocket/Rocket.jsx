import React from 'react';
import './rocket.css';

const Rocket = (props) => {
    const { description, image, name } = props;
    return (
        <div className="rocket_container">
            <img src={image} alt="rockect" />
            <div>
                <h3>{name}</h3>
                <p>{description}</p>
                <button type="button">Reserve Rocket</button>
            </div>
        </div>
    );
}

export default Rocket;