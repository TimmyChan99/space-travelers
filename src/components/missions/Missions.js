import React from 'react';
import missionsDatas from '../../datas/missions';
import Mission from './Mission';
import './missions.css';

export default function Missions() {
  const missions = missionsDatas;

  return (
    <div className="missions-container">
      <div className="header-missions">
        <div className="mission-name-header">Mission</div>
        <div className="mission-description-header">Description</div>
        <div className="mission-status-header">Status</div>
        <div className="mission-other-header">#</div>
      </div>
      {missions.map((mission) => (
        <Mission key={mission.missionId} mission={mission} />
      ))}
    </div>
  );
}
