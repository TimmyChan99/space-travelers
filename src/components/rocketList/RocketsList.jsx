import React from 'react';
import Rocket from '../rocket/Rocket';

const RokectsList = () => {
  const keys = [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }];
  const rockets = keys.map((rocket) => (
    <li key={rocket.id}>
      <Rocket />
    </li>
  ));
  return <ul>{rockets}</ul>;
};

export default RokectsList;
