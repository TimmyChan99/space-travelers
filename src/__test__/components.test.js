import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Missions from '../components/missions/Missions';
import Mission from '../components/missions/Mission';
import Rocket from '../components/rocket/Rocket';
import RocketsList from '../components/rocketList/RocketsList';

it('Test if Missions renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Missions />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

it('Test if Mission renders correctly', () => {
  const mission = {
    missionId: 1,
    missionName: 'test mission name',
    description: 'description',
  };
  const tree = renderer.create(
    <Provider store={store}>
      <Mission mission={mission} />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

it('Test if Rocket renders correctly', () => {
  const item = {
    description: 'description',
    image: 'https://test.com',
    name: 'Rocket',
    id: 1,
    reserved: false,
  };
  const tree = renderer.create(
    <Provider store={store}>
      <Rocket
        description={item.description}
        image={item.image}
        name={item.name}
        id={item.id}
        reserved={item.reserved}
      />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

it('Test if Rockets List renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <RocketsList />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
