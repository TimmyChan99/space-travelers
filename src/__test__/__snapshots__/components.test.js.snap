// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if Mission renders correctly 1`] = `
<div
  className="mission-container"
>
  <div
    className="mission-name"
  >
    test mission name
  </div>
  <div
    className="mission-description"
  >
    description
  </div>
  <div
    className="no-active-mb"
  >
    <span>
       NOT A MEMBER
    </span>
  </div>
  <div
    className="mission"
  >
    <button
      className="join-mission-bt"
      onClick={[Function]}
      type="button"
    >
      Join Mission
    </button>
  </div>
</div>
`;

exports[`Test if Missions renders correctly 1`] = `
<div
  className="missions-container"
>
  <div
    className="header-missions"
  >
    <div
      className="mission-name-header"
    >
      Mission
    </div>
    <div
      className="mission-description-header"
    >
      Description
    </div>
    <div
      className="mission-status-header"
    >
      Status
    </div>
    <div
      className="mission-other-header"
    >
      #
    </div>
  </div>
</div>
`;

exports[`Test if Rocket renders correctly 1`] = `
<div
  className="rocket_container"
>
  <div
    className="image_container"
  >
    <img
      alt="rockect"
    />
  </div>
  <div
    className="info_container"
  >
    <h3 />
    <p />
    <button
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;

exports[`Test if Rockets List renders correctly 1`] = `<ul />`;
